From 84e9bda2599cf4a255cde4ffcad382da21eb8a61 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Fri, 9 Mar 2012 08:51:12 +0100
Subject: [PATCH 3/8] Add support for loading GIO modules from the distro's GIO
 module path

This is needed to e.g. load GVfs from the distro. Without this elements
like giosrc are pretty useless and we really don't want to ship GVfs.
---
 gio/giomodule.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gio/giomodule.c b/gio/giomodule.c
index 510f652..61173fb 100644
--- a/gio/giomodule.c
+++ b/gio/giomodule.c
@@ -1050,6 +1050,11 @@ _g_io_modules_ensure_loaded (void)
 
       g_io_modules_scan_all_in_directory_with_scope (module_dir, scope);
 
+#ifdef GST_SDK_GLIB_GIO_DISTRO_GIO_MODULE_PATH
+      /* Now load all modules from the distro, e.g. gvfs */
+      g_io_modules_scan_all_in_directory_with_scope (GST_SDK_GLIB_GIO_DISTRO_GIO_MODULE_PATH, scope);
+#endif
+
       g_io_module_scope_free (scope);
 
       /* Initialize types from built-in "modules" */
-- 
2.1.0

